[drayton wiser] DraytonWiserMDNSDiscoveryParticipant' threw an exception - ID segment contains invalid characters

Running version 2.5.9 of the binding on openHAB 2.5.9-1 on Ubuntu 18.04.5 LTS, and noted there are frequent errors in openhab.log. I don't believe these are affecting the function of the binding in any way, but would be good if it could be cleared up. Not sure if its an issue specifically with the binding or rather just OH imposing the regex pattern below on the response.

```
2020-10-18 09:53:12.828 [ERROR] [y.mdns.internal.MDNSDiscoveryService] - Participant 'org.openhab.binding.draytonwiser.internal.discovery.DraytonWiserMDNSDiscoveryParticipant' threw an exception
java.lang.IllegalArgumentException: ID segment 'WiserHeat0348EB (14)' contains invalid characters. Each segment of the ID must match the pattern [A-Za-z0-9_-]*.
        at org.eclipse.smarthome.core.common.AbstractUID.validateSegment(AbstractUID.java:97) ~[bundleFile:?]
        at org.eclipse.smarthome.core.common.AbstractUID.<init>(AbstractUID.java:75) ~[bundleFile:?]
        at org.eclipse.smarthome.core.common.AbstractUID.<init>(AbstractUID.java:58) ~[bundleFile:?]
        at org.eclipse.smarthome.core.thing.UID.<init>(UID.java:57) ~[bundleFile:?]
        at org.eclipse.smarthome.core.thing.ThingUID.<init>(ThingUID.java:43) ~[bundleFile:?]
        at org.openhab.binding.draytonwiser.internal.discovery.DraytonWiserMDNSDiscoveryParticipant.getThingUID(DraytonWiserMDNSDiscoveryParticipant.java:86) ~[?:?]
        at org.openhab.binding.draytonwiser.internal.discovery.DraytonWiserMDNSDiscoveryParticipant.createResult(DraytonWiserMDNSDiscoveryParticipant.java:62) ~[?:?]
        at org.eclipse.smarthome.config.discovery.mdns.internal.MDNSDiscoveryService.considerService(MDNSDiscoveryService.java:274) [bundleFile:?]
        at org.eclipse.smarthome.config.discovery.mdns.internal.MDNSDiscoveryService.serviceResolved(MDNSDiscoveryService.java:266) [bundleFile:?]
        at javax.jmdns.impl.ListenerStatus$ServiceListenerStatus.serviceResolved(ListenerStatus.java:102) [bundleFile:3.5.5]
        at javax.jmdns.impl.JmDNSImpl$1.run(JmDNSImpl.java:894) [bundleFile:3.5.5]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_265]
        at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_265]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_265]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_265]
        at java.lang.Thread.run(Thread.java:748) [?:1.8.0_265]
```

--- a/bundles/org.openhab.binding.draytonwiser/src/main/java/org/openhab/binding/draytonwiser/internal/discovery/DraytonWiserMDNSDiscoveryParticipant.java
+++ b/bundles/org.openhab.binding.draytonwiser/src/main/java/org/openhab/binding/draytonwiser/internal/discovery/DraytonWiserMDNSDiscoveryParticipant.java
@@ -19,6 +19,7 @@ import java.util.Collections;
 import java.util.HashMap;
 import java.util.Map;
 import java.util.Set;
+import java.util.regex.Pattern;
 
 import javax.jmdns.ServiceInfo;
 
@@ -45,6 +46,7 @@ import org.slf4j.LoggerFactory;
 public class DraytonWiserMDNSDiscoveryParticipant implements MDNSDiscoveryParticipant {
 
     private final Logger logger = LoggerFactory.getLogger(DraytonWiserMDNSDiscoveryParticipant.class);
+    private final Pattern findIllegalChars = Pattern.compile("[^A-Za-z0-9_-]");
 
     @Override
     public Set<ThingTypeUID> getSupportedThingTypeUIDs() {
@@ -69,10 +71,12 @@ public class DraytonWiserMDNSDiscoveryParticipant implements MDNSDiscoveryPartic
                 if (addresses.length > 0 && addresses[0] != null) {
                     properties.put(PROP_ADDRESS, addresses[0].getHostAddress());
                     properties.put(REFRESH_INTERVAL, DEFAULT_REFRESH_SECONDS);
+
+                    return DiscoveryResultBuilder.create(uid).withProperties(properties)
+                            .withRepresentationProperty(PROP_ADDRESS).withLabel("Heat Hub - " + service.getName())
+                            .build();
                 }
 
-                return DiscoveryResultBuilder.create(uid).withProperties(properties)
-                        .withRepresentationProperty(PROP_ADDRESS).withLabel("Heat Hub - " + service.getName()).build();
             }
         }
         return null;
@@ -83,7 +87,7 @@ public class DraytonWiserMDNSDiscoveryParticipant implements MDNSDiscoveryPartic
         if (service.getType() != null && service.getType().equals(getServiceType())
                 && service.getName().contains("WiserHeat")) {
             logger.trace("Discovered a Drayton Wiser Heat Hub thing with name '{}'", service.getName());
-            return new ThingUID(THING_TYPE_BRIDGE, service.getName());
+            return new ThingUID(THING_TYPE_BRIDGE, findIllegalChars.matcher(service.getName()).replaceAll(""));
         }
         return null;
     }

